feat(documentation): Improvements in the required permissions (#86)
trendmicro/cloudone-filestorage-deployment-templates: aws/required-permissions.md

# User permissions for deployment

> Users need to have specific permissions to execute the deployment.The file [deployment-role-policy.json](deployment-role-policy.json) contains the specific permissions that the users needs to have to deploy the cloudformations templates. You can use a broad, permissive policy such as `AdministratorAccess`, and the deployment will work, but we recommend that roles are more restrictive and have only the permissions required.

Reviewer: ```suggestion
Users need to have specific permissions to execute the deployment. The file [deployment-role-policy.json](deployment-role-policy.json) contains the specific permissions that the users need to deploy the cloudformations templates. You can use a broad, permissive policy such as `AdministratorAccess`, and the deployment will work, but we recommend that roles are restricted to only the permissions required.
```

> The user that will make the deployment need to have specific permissions to execute the deployment. In the file [FSS-user-permissions.json](FSS-user-permissions.json) you can find the specific permissions that the user policy need to have to deploy the cloudformations templates. If your current user already have a more permissive policy such as `AdministratorAccess` the deployment will work, but we do recommend to apply the principle of least privilege.

Reviewer 2: `FSS-user-permissions.json` to `deployment-role-policy.json`

To add the policy:

> - Go to your `AWS account > IAM > Policies > Create policy`

Reviewer: ```suggestion
- Go to `AWS account > IAM > Policies > Create policy`
```

> - Select `JSON` and copy and paste the whole content of the [FSS-user-permissions.json](FSS-user-permissions.json) file.

Reviewer 2: `FSS-user-permissions.json` to `deployment-role-policy.json`

Author: Done!

> - Add `Tags` to the policy (optional), give a `Name` to the policy (Required) and a `Description` (Optional). Your policy is ready to be use!

Reviewer: ```suggestion
- Add `Tags` to the policy (optional), give a `Name` to the policy (Required) and a `Description` (Optional). Your policy is ready to be use.
```

To attach the policy to the user:

> - Go to your desired user on `IAM > Users`

Reviewer: ```suggestion
- Go to `IAM > Users` and select your user.
```

> - Click on `Add permissions`

Reviewer: ```suggestion
- Click `Add permissions`
```

> - Select `Attach existing policies directly`

Reviewer: ```suggestion
- Select `Attach existing policies directly`.
```

> - Mark the policy that you just created and click on `Next > Add permission`.

Reviewer: ```suggestion
- Select the policy that you just created and click on `Next > Add permission`.
```

> - Mark the polcy that we've just created and click on `Next > Add permission`.

Reviewer: ```suggestion
- Mark the polcy that you just created and click on `Next > Add permission`.
```

# Permissions required by the templates

> The templates will create a few IAM roles and policies in order to work properly.

Reviewer: ```suggestion
The templates will create the required IAM roles and policies in order to work properly.
```
